sequencer.c

```diff
@@ -5603,10 +5605,101 @@ static int skip_unnecessary_picks(struct repository *r,
 	return 0;
 }
 
+struct todo_add_branch_context {
+	struct todo_list new_list;
+	struct strbuf *buf;
+};
+
+static int add_branch_for_decoration(const struct name_decoration *d, void *data)
+{
+	struct todo_add_branch_context *ctx = data;
+	size_t base_offset = ctx->buf->len;
+	int i = ctx->new_list.nr;
+	struct todo_item *item;
+	const char *branch_name;
+
+	/* Do not try updating a branch that is checked out. */
+	if (branch_checked_out(d->name, NULL))
+		return 0;
+
+	ALLOC_GROW(ctx->new_list.items,
+		   ctx->new_list.nr + 1,
+		   ctx->new_list.alloc);
+
+
+	if (!skip_prefix(d->name, "refs/heads/", &branch_name))
+		branch_name = d->name;
+
+	strbuf_addf(ctx->buf, "git branch -f %s\n", branch_name);
```

Comment:
This is something I frequently do over here, but I do it in `vi`, transforming the todo list by adding commands of the form:

```sh
git update-ref <ref> HEAD <original-SHA>
```

Maybe we could do the same here?

Reply:
ok, sure.